extensions/wikia/PortableInfobox/services/Helpers/InfoboxTagAttrValidator.php

```diff
@@ -0,0 +1,26 @@
+<?php
+namespace Wikia\PortableInfobox\Helpers;
+
+class InfoboxTagAttrValidator {
+	private $supportedAttributes = [
+		'theme',
+		'theme-source',
+		'layout'
+	];
+
+	/**
+	 * validates infobox tags attribute names
+	 * @param array $attributes
+	 * @throws notValidInfoboxAttributesErrorException
+	 */
+	public function validateAttributes( $attributes ) {
+		foreach ( array_keys( $attributes ) as $attr ) {
+			if ( !in_array( $attr, $this->supportedAttributes ) ) {
```

Comment:
For such a short list optimization would be an overkill; Once we grow the validator we may want to consider using a hashmap instead of array and checking isset($this->supportedAttributes[$attr]).
Feel free to add a note in a comment if it makes sense :)